skbio/tree/_tree.py

```diff
@@ -3269,3 +3269,295 @@ def unpack_by_func(self, func):
                 nodes_to_unpack.append(node)
         for node in nodes_to_unpack:
             node.unpack()
+
+    @experimental(as_of="0.5.3")
+    def is_ultrametric(self):
+        """Returns whether the tree is ultrametric or not.
+
+        Returns
+        -------
+        bool
+            True is all root-to-tip distance are equal, or False if at least
+            one single distance differs
+
+        Notes
+        -----
+        This function collects all the tip to root branch lengths in a set.
+        If this set has a length of 1, then the tree is ultrametric.
+
+        Raises
+        ------
+        NoLengthError
+            A NoLengthError will be thrown if a node without length is
+            encountered
+
+        See Also
+        --------
+        distance
+        tip_tip_distances
+
+        Examples
+        --------
+        >>> from six import StringIO
+        >>> from skbio import TreeNode
+        >>> # example for a non-ultrametric tree
+        >>> tree = TreeNode.read(["((a:1,b:2)c:3,(d:4,e:5)f:6)root;"])
+        >>> tree.is_ultrametric()
+        False
+        >>> tip_root = tree.root()
+        >>> print([tip_root.distance(tree.find(tip)) for tip in tree.tips()])
+        [4.0, 5.0, 10.0, 11.0]
+        >>> # example for an ultrametric tree
+        >>> tree = TreeNode.read(['((a:3,b:3)e:7,(c:2,d:2)f:8);'])
+        >>> tree.is_ultrametric()
+        True
+        >>> tip_root = tree.root()
+        >>> print([tip_root.distance(tree.find(tip)) for tip in tree.tips()])
+        [10.0, 10.0, 10.0, 10.0]
+        """
+        tip_root = self.root()
+        metrics = set()
+        for tip in self.tips():
+            root_to_tip_dist = tip_root.distance(self.find(tip))
```

Comment:
why are you using `self.find(tip)` instead of directly `tip`? What if there are multiple tips with the same name? Not sure what `find` returns in these ambiguous situations but you probably don't want to rely on tip names (they might be None as well).